TestBots/Bridge/PBN.cs

```diff
@@ -10,33 +10,44 @@ internal class PBN
     {
         private const string Sides = "NESW";
         private const string SuitLetters = "SHDC";
+        private const string CardRanks = " 23456789TJQKA";
+
+        public static int CardRank(string card)
+        {
+            return CardRanks.IndexOf(card[1]);
+        }
 
         public static BasicTests.BasicTest[] ImportTests(string text)
         {
             var tests = new List<BasicTests.BasicTest>();
+            var contract = "";
+            var dealerSeat = 0;
             var hands = new List<string>();
+            var history = new List<string>();
             var tags = TokenizeTags(text);
             var name = "";
 
             foreach (var tag in tags)
+            {
                 switch (tag.Name)
                 {
                     case "Event":
                         name = tag.Description;
                         break;
                     case "Deal":
-                        hands = ImportHands(tag.Description);
+                        dealerSeat = Sides.IndexOf(tag.Description.Substring(0, 1).ToUpper());
+                        hands = ImportHands(dealerSeat, tag.Description);
                         break;
                     case "Auction":
                     {
-                        var dealerSide = Sides.IndexOf(tag.Description);
+                        dealerSeat = Sides.IndexOf(tag.Description.ToUpper());
                         var bids = ImportBids(tag.Data);
-                        var history = new List<string>();
+                        history = new List<string>();
                         for (var i = 0; i < bids.Count; i++)
                         {
                             var bid = bids[i];
-                            var seat = i % 4;
-                            var seatName = Sides[(4 - dealerSide + seat) % 4];
+                            var seat = (dealerSeat + i) % 4;
+                            var seatName = Sides[seat];
```

Comment:
@RalphLipe FYI this fixes an existing bug where we'd mis-label the seat associated with a hand in the test logs if the dealer was anything other than North (if I recall correctly, you mentioned running into this at one point).